components/Footer/index.js

```diff
@@ -0,0 +1,104 @@
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import StyledFooter from './StyledFooter';
+import StyledFooterWrapper from './StyledFooterWrapper';
+import StyledColumnSection from './StyledColumnSection';
+import Img from '../Img';
+import StyledIcon from './StyledIcon';
+import StyledRow from './StyledRow';
+import StyledColumn from './StyledColumn';
+import Anchor from './Anchor';
+
+const getRowIcon = (src, srcHover, alt, styling) => {
+  return (
+    <StyledIcon
+      src={src}
+      srcHover={srcHover}
+      alt={alt}
+      styling={styling || {}}
+      key={src}
+    />
+  )
+}
+
+const handleColumn = section =>
+  <StyledColumn styling={section.styling || {}}>
+    {section.title && <h6>{section.title}</h6>}
+    {section.content.map(content => {
+      //can be an image, link or just text
+      if(content.src && content.linkTo && content.external){
```

Comment:
A good way to break down logic definition with the ternary operator is to see it as a bracket, it opens and it closes. This _might_ be an easier way to structure your code, but depends on how used you are to read ternary operator evaluations. 

You are free to keep or change the code as with this example.

```
const externalSrcContentWithLink = (
  <a key={content.linkTo} href={content.linkTo} target='_blank' rel='noopener noreferrer'>
    {getRowIcon(content.src, content.srcHover, content.alt, content.styling)}
  </a>

// Other components go here

const componentToRender = 
  content.src ? 
    content.external ? 
      content.linkTo ? 
        externalSrcContentWithLink :
        internalSrcContentWithLink
      : 
      srcContent
   :
   content.external ? 
      content.linkTo ? 
        externalNameContentWithLink :
        internalNameContentWithLink
      : 
      nameContent
```